R/corrpart.b.R

```diff
@@ -246,72 +233,59 @@ corrPartClass <- R6::R6Class(
 
             return(data)
         },
-        .test = function(var1, var2, controls, type) {
+        .test = function(var1, var2, controls, type, hyp) {
 
             data <- private$.cleanData(var1, var2)
             var1 <- data[[var1]]
             var2 <- data[[var2]]
 
-            if (type != 'zero' && length(controls) > 0)
-                controls <- data[, controls]
-
-            results <- list()
+            n <- length(data[[1]])
+            d <- 1 + as.integer(type == 'semi')
+            c <- length(controls)
+            if (c == 0) { rdta <- cbind(var1, var2) } else { rdta = cbind(var1, var2, data[, controls]) }
+            results <- replicate(d, list(r = NaN, rp = 1, rho = NaN, rhop = 1, tau = NaN, taup = 1, n = n))
```

Comment:
I don't completely get why you make two lists, only if `type == 'semi'`. As they are mutually exclusive you only need to do `semi` or `part` correlations no and thus there's no need to store both. But maybe I'm missing something.